src/Aspire.Hosting.Azure.AIFoundry/AzureAIFoundryExtensions.cs

```diff
@@ -0,0 +1,360 @@
+    /// <param name="modelName">The name of the model to deploy.</param>
+    /// <param name="modelVersion">The version of the model to deploy.</param>
+    /// <param name="format">The format of the model to deploy.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureAIFoundryDeploymentResource> AddDeployment(this IResourceBuilder<AzureAIFoundryResource> builder, [ResourceName] string name, string modelName, string modelVersion, string format)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+        ArgumentException.ThrowIfNullOrEmpty(modelName);
+        ArgumentException.ThrowIfNullOrEmpty(modelVersion);
+        ArgumentException.ThrowIfNullOrEmpty(format);
+
+        var deployment = new AzureAIFoundryDeploymentResource(name, modelName, modelVersion, format, builder.Resource);
+
+        builder.ApplicationBuilder.AddResource(deployment);
+
+        builder.Resource.AddDeployment(deployment);
+
+        var deploymentBuilder = builder.ApplicationBuilder
+            .CreateResourceBuilder(deployment);
+
+        if (builder.Resource.IsEmulator)
+        {
+            deploymentBuilder.AsLocalDeployment(deployment);
+        }
+
+        return deploymentBuilder;
+    }
+
+    /// <summary>
+    /// Allows setting the properties of an Azure AI Foundry Deployment resource.
+    /// </summary>
+    /// <param name="builder">The Azure AI Foundry Deployment resource builder.</param>
+    /// <param name="configure">A method that can be used for customizing the <see cref="AzureAIFoundryDeploymentResource"/>.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureAIFoundryDeploymentResource> WithProperties(this IResourceBuilder<AzureAIFoundryDeploymentResource> builder, Action<AzureAIFoundryDeploymentResource> configure)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(configure);
+
+        configure(builder.Resource);
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a Foundry Local resource to the distributed application builder.
+    /// </summary>
+    /// <param name="builder">The distributed application builder.</param>
+    /// <returns>A resource builder for the Foundry Local resource.</returns>
+    public static IResourceBuilder<AzureAIFoundryResource> RunAsFoundryLocal(this IResourceBuilder<AzureAIFoundryResource> builder)
+    {
+        ArgumentNullException.ThrowIfNull(builder, nameof(builder));
+
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return builder;
+        }
+
+        var resource = builder.Resource;
+        resource.Annotations.Add(new EmulatorResourceAnnotation());
+
+        var resourceBuilder = builder.ApplicationBuilder
```

Comment:
Why do you need a new ResourceBuilder? We already have one that is passed in.

Reply:
Left-over from when I was creating a new resource for the local foundry. Fixed now.